include/boost/math/tools/big_constant.hpp

```diff
@@ -33,9 +33,12 @@ struct numeric_traits<__float128>
    static const int max_exponent = 16384;
    static const bool is_specialized = true;
 };
-#else
+#elif LDBL_DIG > DBL_DIG
 typedef long double largest_float;
 #define BOOST_MATH_LARGEST_FLOAT_C(x) x##L
+#else
+typedef double largest_float;
+#define BOOST_MATH_LARGEST_FLOAT_C(x) x
```

Comment:
Surely this breaks long double support for those that want it?

Reply:
This `else` condition is met when double and long double are identical (`LDBL_DIG == DBL_DIG`) - so either would qualify as the largest float on those platforms. I do not believe this will have any run-time effect or break any compiles but I could be wrong (maybe if the result of `BOOST_MATH_LARGEST_FLOAT_C` is passed to a templated function?).

I'm basically indifferent to this particular change - but it does remove the last of the `long double`s from my debug symbols.

Reply:
Ah yes, my bad: I was viewing the diff and the PP logic got a bit confused at least in my head! ;)